docs/source/reference/deprecation.rst

```diff
@@ -0,0 +1,327 @@
+===================
+Deprecation Notices
+===================
+
+This section contains information about deprecation of behaviours, features and
+APIs that have become undesirable/obsolete. Any information about the schedule
+for their deprecation and reasoning behind the changes, along with examples, is
+provided. However, first is a small section on how to suppress deprecation
+warnings that may be raised from Numba so as to prevent warnings propagating
+into code that is consuming Numba.
+
+Suppressing Deprecation warnings
+================================
+All Numba deprecations are issued via ``NumbaDeprecationWarning`` or
+``NumbaPendingDeprecationWarning`` s, to suppress the reporting of
+these the following code snippet can be used::
+
+    from numba.errors import NumbaDeprecationWarning, NumbaPendingDeprecationWarning
+    import warnings
+
+    warnings.simplefilter('ignore', category=NumbaDeprecationWarning)
+    warnings.simplefilter('ignore', category=NumbaPendingDeprecationWarning)
+
+The ``action`` used above is ``'ignore'``, other actions are available, see
+`The Warnings Filter <https://docs.python.org/3/library/warnings.html#the-warnings-filter>`_
+documentation for more information.
```

Comment:
This might also be a good place to offer the blanket recommendation that applications and libraries which choose to suppress warnings should also pin their Numba dependency because their users will no longer be aware of the coming incompatibility.

Reply:
Good point. I've added a `.. note::` to emphasise this.